scarb/src/compiler/compilers/lib.rs

```diff
@@ -60,11 +61,15 @@ impl Compiler for LibCompiler {
 
         if props.sierra {
             let mut file = target_dir.open_rw(
-                format!("{}.sierra", unit.target().name),
+                format!("{}.sierra.json", unit.target().name),
                 "output file",
                 ws.config(),
             )?;
-            file.write_all(sierra_program.to_string().as_bytes())?;
+            file.write_all(
+                serde_json::to_string_pretty(&VersionedProgram::from((*sierra_program).clone()))
+                    .context("failed to serialize sierra versioned program")?
+                    .as_bytes(),
+            )?;
```

Comment:
1. It's ineffective to generate text to string and then write it to file. You can do that in one operation: https://docs.rs/serde_json/latest/serde_json/fn.to_writer.html
2. `starknet-contract` generates minified JSON. Let's do that here too.